lib/streamlit/elements/widgets/button_group.py

```diff
@@ -300,76 +379,161 @@ def feedback(
+        style: Literal["segment", "pills"] = "segment",
         on_change: WidgetCallback | None = None,
         args: WidgetArgs | None = None,
         kwargs: WidgetKwargs | None = None,
-    ) -> list[V]:
-        def _transformed_format_func(x: V) -> ButtonGroupProto.Option:
-            if format_func is None:
-                return ButtonGroupProto.Option(content=str(x))
-
-            transformed = format_func(x)
+    ) -> list[V] | V | None:
+        def _transformed_format_func(option: V) -> ButtonGroupProto.Option:
+            """If option starts with a material icon or an emoji, we extract it to send
+            it parsed to the frontend."""
+            transformed = format_func(option) if format_func else str(option)
+            transformed_parts = transformed.split(" ")
+            icon: str | None = None
+            if len(transformed_parts) > 0:
+                maybe_icon = transformed_parts[0].strip()
+                try:
+                    icon = validate_icon_or_emoji(maybe_icon)
+                    # reassamble the option string without the icon - also
+                    # works if len(transformed_parts) == 1
+                    transformed = " ".join(transformed_parts[1:])
+                except StreamlitAPIException:
+                    # we don't have a valid icon or emoji, so we just pass
+                    pass
             return ButtonGroupProto.Option(
-                content=transformed["content"],
-                selected_content=transformed["selected_content"],
+                content=transformed,
+                content_icon=icon,
             )
 
         indexable_options = convert_to_sequence_and_check_comparable(options)
         default_values = get_default_indices(indexable_options, default)
-        serde = MultiSelectSerde(indexable_options, default_values)
+
+        serde: SingleOrMultiSelectSerde[V] = SingleOrMultiSelectSerde[V](
+            indexable_options, default_values, selection_mode
+        )
 
         res = self._button_group(
             indexable_options,
             key=key,
             default=default_values,
-            click_mode=ButtonGroupProto.ClickMode.MULTI_SELECT
-            if click_mode == "multiselect"
-            else ButtonGroupProto.SINGLE_SELECT,
+            selection_mode=selection_mode,
             disabled=disabled,
             format_func=_transformed_format_func,
+            style=style,
             serializer=serde.serialize,
             deserializer=serde.deserialize,
             on_change=on_change,
             args=args,
             kwargs=kwargs,
-            after_register_callback=lambda widget_state: maybe_coerce_enum_sequence(
-                widget_state, options, indexable_options
-            ),
         )
-        return res.value
+
+        if selection_mode == "multi":
+            return res.value
+
+        return res.value if selection_mode == "single" else None
```

Comment:
Same comment as above: I think we can just do: `return res.value`

Reply:
Removed it 👍